src/project-type/workspace-instance.ts

```diff
@@ -1,18 +1,38 @@
-import { WorkspaceApi } from "@sap/project-api";
-
-// TODO: define subset of properties signatures
-let workspaceAPIProxy: WorkspaceApi;
+import { ProjectApi, Tag, WorkspaceApi } from "@sap/project-api";
+import { Uri, WorkspaceFolder } from "vscode";
+import { BasWorkspaceApi } from "../../types/api";
 
+let basWSAPI: BasWorkspaceApi;
 /**
  * @param WorkspaceImpl - constructor for the `WorkspaceApi`
  *                        dependency injection is used to enable easier testing
  */
-export function initWorkspaceAPI(WorkspaceImpl: { new (): WorkspaceApi }) {
-  workspaceAPIProxy = new WorkspaceImpl();
-  // TODO: implement "READ-ONLY" proxy
+export function getBasWorkspaceAPI(WorkspaceImpl: { new(): WorkspaceApi }): BasWorkspaceApi {
```

Comment:
The split of the `init` and the `getter` sounds logical.
But because of the DI for the constructor it means everyone who calls `getBasWorkspaceAPI`
now needs to know how to provide the constructor so the end result may be worse.

Reply:
It also prevents us from ever changing the `basWSAPI` instance (e.g in testing)
As with this code it would only be initialized **once**.

Reply:
Done